File: Java/TrocaDeVariaveis.java
public class TrocaDeVariaveis {
	public static void main(String[] args) {
		int a = 10;
		int b = 20;
        System.out.println("A = " + a);
		System.out.println("B = " + b);
       	System.out.println("+++++++");

		int aux = a;
		a = b;
		b = aux;

		System.out.println("A = " + a);
		System.out.println("B = " + b);
	}
}

File: JavaScript/TrocaDeVariaveis.js
/**
 * Algoritmo de Troca de Variáveis
 * Demonstra a troca de valores entre duas variáveis utilizando uma variável auxiliar.
 */

function executarTroca() {
    let a = 10;
    let b = 20;
    console.log(`A = ${a}`);
    console.log(`B = ${b}`);
    console.log('+++++++');

    const aux = a;
    a = b;
    b = aux;

    console.log(`A = ${a}`);
    console.log(`B = ${b}`);

    return { a, b };
}

if (require.main === module) {
    executarTroca();
}

module.exports = {
    executarTroca,
};
